macros/one_hot_encoder.sql

```diff
@@ -1,56 +1,89 @@
+    {%- endif -%}
+    {%- if include_columns!='*' or exclude_columns is not none -%}
+        {%- if include_columns!='*' and exclude_columns is not none -%}
+            {% set error_message %}
+        The `one_hot_encoder` macro only supports one of include_columns and exclude_columns being a non-default value or one of them being a list.
+            {% endset %}
+            {%- do exceptions.raise_compiler_error(error_message) -%}
+        {%- elif include_columns!='*' and (include_columns is not iterable  or include_columns is string or include_columns is mapping) -%}
+            {% set error_message %}
+        The `one_hot_encoder` macro only supports one of include_columns and exclude_columns being a non-default value or one of them being a list.
+            {% endset %}
+            {%- do exceptions.raise_compiler_error(error_message) -%}
+        {%- elif exclude_columns is not none and (exclude_columns is not iterable or exclude_columns is string or exclude_columns is mapping) -%}
+                    {% set error_message %}
+        The `one_hot_encoder` macro only supports one of include_columns and exclude_columns being a non-default value or one of them being a list.
+            {% endset %}
+            {%- do exceptions.raise_compiler_error(error_message) -%}
+        {%- endif -%}
+    {%- endif -%}
+    {{ adapter.dispatch('one_hot_encoder',packages=['dbt_ml_preprocessing'])(source_table, source_column, category_values, handle_unknown, include_columns, exclude_columns) }}
 {%- endmacro %}
 
-{% macro default__one_hot_encoder(source_table,source_column,category_values,handle_unknown,include_columns) %}
-select 
-{% for column in include_columns %}
-{{ column }},
-{% endfor %}
-{% for category in category_values %}
-case {{source_column}}
-    when '{{category}}' then true
-    else false
-    end as {{source_column}}_{{category}}
-{% if not loop.last %}, {% endif %}
-{% endfor %}
-from {{ source_table }}
+{% macro default__one_hot_encoder(source_table, source_column, category_values, handle_unknown, include_columns, exclude_columns) %}
+    {% set columns = adapter.get_columns_in_relation( source_table ) %}
+
+
+
+
+    with binary_output as (
+    select
+        {%- if include_columns=='*' and exclude_columns is  none -%}
+            {% for column in columns %}
+                {{ column.name }},
+            {%- endfor -%}
+        {%- elif include_columns !='*'-%}
+            {% for column in include_columns %}
+                {{ source_table }}.{{ column }},
+            {%- endfor -%}
+        {%- else -%}
+            {% for column in columns %}
+            {%- if column.name | lower not in exclude_columns | lower %}
+                {{ column.name }},
+            {%- endif -%}
+            {%- endfor -%}
+        {%- endif -%}
+        {% for category in category_values %}
+            {% set no_whitespace_column_name = category | replace( " ", "_") -%}
+                {%- if category is none %}
```

Comment:
`null` types in SQL are `none` types in Jinja, so slight change in approach for the case where the value is `null`